shared-libs/user-management/test/unit/users.spec.js

```diff
@@ -190,186 +196,296 @@ describe('Users service', () => {
+        const usersByIds = [
+          {
+            _id: 'org.couchdb.user:x',
             name: 'lucas',
             facility_id: 'c',
-            roles: [ 'national-admin', 'data-entry' ]
-          }
-        },
-        {
-          id: 'org.couchdb.user:y',
-          doc: {
-            name: 'milan',
-            facility_id: 'b',
-            roles: [ 'district-admin' ]
-          }
-        }
-      ];
-      const allUsersSettings = [
-        {
-          _id: 'org.couchdb.user:x',
-          name: 'lucas',
-          fullname: 'Lucas M',
-          email: 'l@m.com',
-          phone: '123456789'
-        },
-        {
-          _id: 'org.couchdb.user:y',
-          name: 'milan',
-          fullname: 'Milan A',
-          email: 'm@a.com',
-          phone: '987654321',
-          external_id: 'LTT093'
-        }
-      ];
-      service.__set__('getAllUsers', sinon.stub().resolves(allUsers));
-      service.__set__('getAllUserSettings', sinon.stub().resolves(allUsersSettings));
-      return service.getList().then(data => {
-        chai.expect(data.length).to.equal(2);
-        const lucas = data[0];
-        chai.expect(lucas.id).to.equal('org.couchdb.user:x');
-        chai.expect(lucas.username).to.equal('lucas');
-        chai.expect(lucas.fullname).to.equal('Lucas M');
-        chai.expect(lucas.email).to.equal('l@m.com');
-        chai.expect(lucas.phone).to.equal('123456789');
-        chai.expect(lucas.place).to.deep.equal(facilityc);
-        chai.expect(lucas.roles).to.deep.equal([ 'national-admin', 'data-entry' ]);
-        const milan = data[1];
-        chai.expect(milan.id).to.equal('org.couchdb.user:y');
-        chai.expect(milan.username).to.equal('milan');
-        chai.expect(milan.fullname).to.equal('Milan A');
-        chai.expect(milan.email).to.equal('m@a.com');
-        chai.expect(milan.phone).to.equal('987654321');
-        chai.expect(milan.place).to.deep.equal(facilityb);
-        chai.expect(milan.roles).to.deep.equal([ 'district-admin' ]);
-        chai.expect(milan.external_id).to.equal('LTT093');
-      });
-    });
-
-    it('filters out non-users', () => {
-      const allUsers = [
-        {
-          id: 'x',
-          doc: {
+            roles: ['national-admin', 'data-entry']
+          },
+        ];
+        const allUsersSettings = [
+          {
+            _id: 'org.couchdb.user:x',
             name: 'lucas',
-            facility_id: 'c',
             fullname: 'Lucas M',
             email: 'l@m.com',
-            phone: '123456789',
-            roles: [ 'national-admin', 'data-entry' ]
-          }
-        },
-        {
-          id: 'org.couchdb.user:y',
-          doc: {
+            phone: '123456789'
+          },
+        ];
+        const getUsersByIds = sinon.stub().resolves(usersByIds);
+        service.__set__('getAllUserSettings', sinon.stub().resolves(allUsersSettings));
+        service.__set__('getUsersByIds', getUsersByIds);
```

Comment:
We have a good bit of logic in these functions that AFICT is not getting unit tested at all since we are stubbing these out.  

Would it be possible to update your new "with filters" tests to do the stubbing at the db level so these tests can cover the actual behavior of these functions?